rpcs3/rpcs3qt/game_compatibility.cpp

```diff
@@ -4,22 +4,59 @@
 
 #include <QMessageBox>
 #include <QJsonDocument>
-#include <QNetworkReply>
+
+#include <thread>
+
+#include <curl/curl.h>
 
 LOG_CHANNEL(compat_log, "Compat");
 
 constexpr auto qstr = QString::fromStdString;
 inline std::string sstr(const QString& _in) { return _in.toStdString(); }
 
+size_t curl_write_cb_compat(char* ptr, size_t size, size_t nmemb, void* userdata)
+{
+	game_compatibility* gm_cmp = reinterpret_cast<game_compatibility*>(userdata);
+	return gm_cmp->update_buffer(ptr, nmemb);
+}
+
 game_compatibility::game_compatibility(std::shared_ptr<gui_settings> settings) : m_xgui_settings(settings)
 {
 	m_filepath = m_xgui_settings->GetSettingsDir() + "/compat_database.dat";
 	m_url = "https://rpcs3.net/compatibility?api=v1&export";
-	m_network_request = QNetworkRequest(QUrl(m_url));
+
+	m_curl = curl_easy_init();
 
 	RequestCompatibility();
 }
 
+size_t game_compatibility::update_buffer(char* data, size_t size)
+{
+	if (m_curl_abort)
+	{
+		return 0;
+	}
+
+	auto old_size = m_curl_buf.size();
+	auto new_size = old_size + size;
+	m_curl_buf.resize(new_size);
+	memcpy(m_curl_buf.data() + old_size, data, size);
+
+	if (!actual_dwnld_size)
+	{
+		if (curl_easy_getinfo(m_curl, CURLINFO_CONTENT_LENGTH_DOWNLOAD, &actual_dwnld_size) != -1)
```

Comment:
The docs don't mention any CURLcode with value of -1, are you sure this comparison is correct?